File: Java-SpringBoot/BE-02-DB/java-17-springboot-3.5/src/main/java/com/flyrank/taskservice/TaskServiceApplication.java
package com.flyrank.taskservice;

import com.flyrank.taskservice.entity.TaskEntity;
import com.flyrank.taskservice.repository.TaskRepository;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

import java.util.List;

@SpringBootApplication
public class TaskServiceApplication {

    public static void main(String[] args) {
        SpringApplication.run(TaskServiceApplication.class, args);
    }

    /**
     * Seeds initial sample task data if the persistent database is empty.
     */
    @Bean
    public CommandLineRunner seedDatabase(TaskRepository repository) {
        return args -> {
            if (repository.count() == 0) {
                repository.saveAll(List.of(
                        new TaskEntity("Buy groceries", false),
                        new TaskEntity("Complete Flyrank assignment BE-02", true),
                        new TaskEntity("Review Spring Boot 4.0 release features", false)
                ));
            }
        };
    }
}

File: Java-SpringBoot/BE-02-DB/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/taskservice/TaskServiceApplication.kt
package com.flyrank.taskservice

import com.flyrank.taskservice.entity.TaskEntity
import com.flyrank.taskservice.repository.TaskRepository
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean

@SpringBootApplication
class TaskServiceApplication {

    @Bean
    fun seedDatabase(repository: TaskRepository) = CommandLineRunner {
        if (repository.count() == 0L) {
            repository.saveAll(
                listOf(
                    TaskEntity(title = "Buy groceries", done = false),
                    TaskEntity(title = "Complete Flyrank assignment BE-02 in Kotlin", done = true),
                    TaskEntity(title = "Explore Kotlin 2.3 & Spring Data JPA", done = false)
                )
            )
        }
    }
}

fun main(args: Array<String>) {
    runApplication<TaskServiceApplication>(*args)
}
